pkg/controller/elasticsearch/driver/nodes.go

```diff
@@ -124,6 +124,10 @@ func (d *defaultDriver) reconcileNodeSpecs(
 		return results.WithError(err)
 	}
 
+	if err := reconcilePVCOwnerRefs(d.K8sClient(), d.ES); err != nil {
+		return results.WithError(err)
+	}
```

Comment:
let's reconcile owner refs before we garbage collect pvcs?
Otherwise I get funny error messages because we update owner refs after the pvc was deleted:

```
2021-03-02T16:58:56.354+0100	INFO	driver	Deleting PVC	{"service.version": "1.5.0-SNAPSHOT+45c993aa", "namespace": "default", "pvc_name": "elasticsearch-data-elasticsearch-sample-es-default-1"}
2021-03-02T16:58:56.438+0100	INFO	elasticsearch-controller	Ending reconciliation run	{"service.version": "1.5.0-SNAPSHOT+45c993aa", "iteration": 21, "namespace": "default", "es_name": "elasticsearch-sample", "took": "773.970731ms"}
2021-03-02T16:58:56.438+0100	ERROR	manager.eck-operator.controller.elasticsearch-controller	Reconciler error	{"service.version": "1.5.0-SNAPSHOT+45c993aa", "name": "elasticsearch-sample", "namespace": "default", "error": "while updating pvc during owner ref reconciliation: Operation cannot be fulfilled on persistentvolumeclaims \"elasticsearch-data-elasticsearch-sample-es-default-1\": StorageError: invalid object, Code: 4, Key: /registry/persistentvolumeclaims/default/elasticsearch-data-elasticsearch-sample-es-default-1, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: cd93c74e-ed72-4661-8617-7f4f8d7b8cc4, UID in object meta: ", "errorCauses": [{"error": "while updating pvc during owner ref reconciliation: Operation cannot be fulfilled on persistentvolumeclaims \"elasticsearch-data-elasticsearch-sample-es-default-1\": StorageError: invalid object, Code: 4, Key: /registry/persistentvolumeclaims/default/elasticsearch-data-elasticsearch-sample-es-default-1, ResourceVersion: 0, AdditionalErrorMsg: Precondition failed: UID in precondition: cd93c74e-ed72-4661-8617-7f4f8d7b8cc4, UID in object meta: "}]}
```

Reply:
Good point. I initially thought GC first made sense as it would whittle down the list of PVCs that need ownerRef updates but with a cached client that does not make sense and actually causes the problem you saw.